### PR TITLE
Timelock bug fix

### PR DESCRIPTION
**Pull request by nkohen**
Before this PR, the test case that is added here fails.

The problem was that the current default of `UInt32.zero` was being interpreted as a block height and since BIP 65 does not allow block height timelocks alongside timestamp ones this caused an error any time anyone tried using a second-timestamp timelock.

I also ran `test:scalafmt` and it seems a couple other files were hit by that in the diff

### CODE REVIEW

File: core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala [hunk 1]
```diff
@@ -410,47 +410,54 @@ sealed abstract class BitcoinTxBuilder extends TxBuilder {
...
-              _: EmptySpendingInfo =>
-            // none of these scripts affect the locktime of a tx
-            loop(newRemaining, currentLockTime)
-        }
-    }
+                    } else Success(UInt32(cltv.locktime.toLong))
+                  val result = lockTime.flatMap { l: UInt32 =>
+                    currentLockTimeOpt match {
+                      case Some(currentLockTime) =>
+                        if (currentLockTime < l) {
```
Comment (benthecarman) [inline core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala:432]: Doesn't this assume that `l` is a block time and not a block height, if so wouldn't this break when a block height is given for `l` and `currentLockTime` is a block time

**Review by benthecarman — APPROVED** [review]

File: core-test/src/test/scala/org/bitcoins/core/wallet/builder/BitcoinTxBuilderTest.scala [hunk 2]
```diff
@@ -387,6 +387,36 @@ class BitcoinTxBuilderTest extends BitcoinSAsyncTest {
...
+        Vector(
+          TransactionOutput(Bitcoins.one - CurrencyUnits.oneMBTC,
+                            EmptyScriptPubKey)),
+        Vector(cltvSpendingInfo),
+        SatoshisPerByte(Satoshis.one),
+        EmptyScriptPubKey,
+        RegTest
+      )
+
+    txBuilderF.flatMap(_.sign).map(_ => succeed)
```
Comment (Christewart) [inline core-test/src/test/scala/org/bitcoins/core/wallet/builder/BitcoinTxBuilderTest.scala]: Shouldn't we verify this in the script interpreter?

File: core-test/src/test/scala/org/bitcoins/core/wallet/builder/BitcoinTxBuilderTest.scala [hunk 3]
```diff
@@ -387,6 +387,36 @@ class BitcoinTxBuilderTest extends BitcoinSAsyncTest {
     }
   }
 
+  it must "succeed to sign a cltv spk that uses a second-based locktime" in {
```
Comment (Christewart) [inline core-test/src/test/scala/org/bitcoins/core/wallet/builder/BitcoinTxBuilderTest.scala:390]: should we add test cases for block height as well? Do you think negative test cases are necessary to?

File: core-test/src/test/scala/org/bitcoins/core/wallet/builder/BitcoinTxBuilderTest.scala (hunk 2 above)
Comment (nkohen) [reply core-test/src/test/scala/org/bitcoins/core/wallet/builder/BitcoinTxBuilderTest.scala]: That shouldn't be necessary as we are only testing `TxBuilder.calcLockTime` here. That said, I did change this to verify that the transaction's `lockTime` is correct rather than just `suceed`ing
